File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/system/get_version.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_system = h10w.H10wSystem()
time.sleep(1)

print("=== 获取机器人各软件的版本号示例 ===")
ret, version = h10w_system.getRobotVersion()

print("Robot Driver Version: ", version.driver_version.driver_version)
print("Robot Firmware Version: ", version.driver_version.firmware_version)

print(
    "Robot Controller Version: ",
    version.controller_version.controller_version,
)
print(
    "Robot Algorithm Version: ",
    version.controller_version.algorithm_version,
)

print("Robot Monitor Main Version: ", version.monitor_version.main_version)
print("Robot Monitor Function Version: ")
for n, v in version.monitor_version.function_versions.items():
    print("\t", n, "version: ", v)

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/system/get_version.cpp
// get_version.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_system.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取机器人各软件的版本号示例 ===" << std::endl;
  H10wSystem system;
  RobotVersion robotVersion;
  system.getRobotVersion(robotVersion);
  std::cout << "驱动器版本号：" << robotVersion.driver_version.driver_version
            << std::endl;
  std::cout << "控制器版本号："
            << robotVersion.controller_version.controller_version << std::endl;
  std::cout << "算法版本号："
            << robotVersion.controller_version.algorithm_version << std::endl;

  std::cout << "安全监控版本号：" << robotVersion.monitor_version.main_version
            << std::endl;

  return 0;
}
